src/mmw/js/src/data_catalog/models.js

```diff
@@ -153,7 +153,17 @@ var Catalog = Backbone.Model.extend({
     },
 
     initialize: function() {
-        var self = this;
+        var self = this,
+            debouncedSearch = _.debounce(function() {
```

Comment:
The commit message is great; a comment might also be helpful on why we're debouncing the search.